node/lib/util/commit.js

```diff
@@ -2222,3 +2222,99 @@ it empty. You can remove the commit entirely with "git meta reset HEAD^".`);
                                        message,
                                        subMessages);
 });
+
+const submoduleHasChangesToCommit = function(status, submoduleName) {
+    const submodule = status.submodules[submoduleName];
+    if (!submodule) {
+        return false;
+    }
+    const { workdir } = submodule;
+    return null !== workdir && !workdir.status.isIndexClean();
+};
+
+const submoduleHasPreCommitHook = co.wrap(function *(repo, submoduleName) {
+    const Hook          = require("../util/hook");
+    const SubmoduleUtil = require("../util/submodule_util");
+
+    const subRepo = yield SubmoduleUtil.getRepo(repo, submoduleName);
+    return Hook.hasHook(subRepo, "pre-commit");
+});
+
+const filterInParallel = co.wrap(function *(collection, asyncPredicate) {
+    const DoWorkQueue = require("../util/do_work_queue");
+
+    const booleans = yield DoWorkQueue.doInParallel(collection,
+        asyncPredicate);
+    return collection.filter((item, index) => booleans[index]);
+});
+
+const getSubmodulesNameForPreCommitCheck = co.wrap(function *(repo,
+                                                              repoStatus) {
+    const SubmoduleUtil = require("../util/submodule_util");
+
+    const allSubmoduleNames = yield SubmoduleUtil.getSubmoduleNames(repo);
+    return yield filterInParallel(
+        allSubmoduleNames,
+        co.wrap(function*(submoduleName) {
+            if (!submoduleHasChangesToCommit(repoStatus, submoduleName)) {
```

Comment:
nit: you could simplify this code by removing the ! and moving ```return yield submoduleHasPreCommitHook(repo, submoduleName); ``` inside the if

Reply:
fixed.